d3a_interface/data_classes.py

```diff
@@ -71,33 +80,41 @@ def to_json_string(self, **kwargs) -> str:
         return json.dumps(obj_dict, default=json_datetime_serializer)
 
     def serializable_dict(self) -> Dict:
+        """Return a json serializable representation of the class."""
         return {
             "type": self.__class__.__name__,
             "id": self.id,
             "energy": self.energy,
             "energy_rate": self.energy_rate,
             "original_price": self.original_price,
             "creation_time": datetime_to_string_incl_seconds(self.creation_time),
+            "time_slot": datetime_to_string_incl_seconds(self.time_slot),
             "attributes": self.attributes,
             "requirements": self.requirements
         }
 
     @classmethod
-    def from_json(cls, offer_or_bid, current_time=None) -> Union["Offer", "Bid"]:
+    def from_json(cls, offer_or_bid: str) -> Union["Offer", "Bid"]:
+        """De-serialize orders from json string."""
         offer_bid_dict = json.loads(offer_or_bid)
         object_type = offer_bid_dict.pop("type")
-
         offer_bid_dict.pop("energy_rate", None)
-
-        offer_bid_dict["creation_time"] = current_time
-
+        if offer_bid_dict.get("creation_time"):
+            offer_bid_dict["creation_time"] = (
+                str_to_pendulum_datetime(offer_bid_dict["creation_time"]))
+        else:
+            offer_bid_dict["creation_time"] = None
+        if offer_bid_dict.get("time_slot"):
+            offer_bid_dict["time_slot"] = str_to_pendulum_datetime(offer_bid_dict["time_slot"])
         if object_type == "Offer":
             return Offer(**offer_bid_dict)
         if object_type == "Bid":
             return Bid(**offer_bid_dict)
+        assert False, "from_json expects a json string containing the 'type' key."
```

Comment:
Nit: this will be triggered also when the JSON string passed contains different values of types (neither "Offer" nor "Bid").